lib/kino_db/sql_cell.ex

```diff
@@ -95,6 +91,17 @@ defmodule KinoDB.SQLCell do
     {:noreply, assign(ctx, connections: connections, connection: connection)}
   end
 
+  defp search_connection([connection | _], nil), do: connection
+
+  defp search_connection([], connection), do: connection
+
+  defp search_connection(connections, %{type: type} = connection) do
+    case Enum.find(connections, &(&1.type == type)) do
```

Comment:
Shouldn't we search by variable name instead?

Reply:
And if we find a connection we return it instead of using `^` below, since we know only the type changed.

Reply:
makes more sense, because a livebook session could have 2 postgres but not with the same variable